apps/fhir/bluebutton/models.py

```diff
@@ -25,27 +25,40 @@ def get_queryset(self):
         return super().get_queryset().filter(Q(_fhir_id__startswith='-'))
 
 
-def hash_hicn(hicn):
+def hash_id_value(hicn):
     """
-    Hashes a hicn to match fhir server logic:
+    Hashes an MBI or HICN to match fhir server logic:
+    Both currently use the same hash salt ENV values.
     https://github.com/CMSgov/beneficiary-fhir-data/blob/master/apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoader.java#L665-L706
-
     """
-    assert hicn != "", "HICN cannot be the empty string"
-
     return binascii.hexlify(pbkdf2(hicn,
                             get_user_id_salt(),
                             settings.USER_ID_ITERATIONS)).decode("ascii")
 
 
+def hash_hicn(hicn):
+    assert hicn != "", "HICN cannot be the empty string"
+
+    return hash_id_value(hicn)
+
+
+def hash_mbi(mbi):
+    assert mbi != "", "MBI cannot be the empty string"
+    # NOTE: mbi value can be None here.
+    if mbi is None:
+        return None
```

Comment:
question: mbi and hicn returned from SLS are hashed or original mbi / hicn? when the identity is from MSLS (local simulator) we usually feed with mbi/hicn hash from synthetic beneficiary records which has both mbi and mbi-hash or/and hicn and hicn-hash, I looked at the  code in mymedicare_cb.models.get_and_update_user(user_info), seems we assume only original mbi and/or hicn are there? but in local env, a fhir search with patient identifier=system|value is used to retrieve the user, where system uri is like: "system":"https://bluebutton.cms.gov/resources/identifier/mbi-hash","value":"0b2b0f53d207a7b5867e834277490fdacf2642fafb51fe40030cc890a73a188f"

Reply:
another question: in mymedicare_cb/models.py below method has user_id_type as argument, but since we explicitly has user_mbi_hash and user_hicn_hash, it seems user_id_type is still needed? 

    user = create_beneficiary_record(username=subject,
                                     user_hicn_hash=hicn_hash,
                                     user_mbi_hash=mbi_hash,
                                     fhir_id=fhir_id,
                                     first_name=first_name,
                                     last_name=last_name,
                                     email=email,
                                     user_id_type=user_id_type)

same question apply to django models Crosswalk.
or I miss understood user_id_type usage?

Reply:
@JFU-GIT The hicn and mbi values from SLS are not hashed yet. We hash them prior to the Patient resource lookup: https://github.com/CMSgov/bluebutton-web-server/blob/800c1963c8b8a9fef362ea128bfca619c90c77e9/apps/mymedicare_cb/models.py#L30

Reply:
@JFU-GIT `user_id_type` is currently always set to HICN. For the next stage of work (next related Jira story), we can use that to denote which type of lookup was used to find the patient ID (HICN/MBI).

Reply:
got it